posts/exploring-iot-infrastructure-support-in-aws-part2.md

```diff
@@ -0,0 +1,450 @@
+---
+authors:
+- Ben Edridge
+tags:
+- AWS
+- IoT
+date: 2018-07-19T11:41:57.000Z
+title: "Exploring AWS IoT Core and Greengrass Offerings (Part 2)"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/07/aws_iot_esp_device.jpeg
+---
+
+This a second post following from the [part one](https://blog.ippon.tech/exploring-iot-infrastructure-support-in-aws-part1) of "Exploring AWS IoT Core and Greengrass Offerings". In part one I discussed the background and theoretical aspects of the AWS IoT offerings including [AWS IoT Core](https://aws.amazon.com/iot-core/), [Greengrass](https://docs.aws.amazon.com/greengrass/latest/developerguide/what-is-gg.html#gg-platforms) and [Amazon FreeRTOS](https://aws.amazon.com/freertos/). I also introduced a proof of concept which will be further discussed in this blog post. More example code will be provided and the deployment options will be evaluated.
+
+# The Journey
+
+It's always been a hobby of mine to explore low-powered/mobile computing devices and hack around. Whether it was revitalising old routers, increasing page limits for printers or customising the Android OS. The basic goal for me was I wanted to have some fun whilst learning about the software and the supporting OS. The questions I always had were:
+
+> Can we get a shell? maybe we can install a custom Linux?
+> We have Linux... Are there ways in which we can give the device more capability?
+
+Surprisingly a huge number of devices run embedded Linux/Unix. I'm not talking about the standard devices I mentioned before. But devices you would never expect like music players, fridges, cars, coffee machines and drones. Many of these make use of open source software/drivers with additional proprietary components.
+
+Over the last few months I purchased a selection of Espressif based IoT devices from the Amazon of China: [Taobao.com](taobao.com) for experimentation, research and potential use in coding Dojos or teaching. Below is an example of one of the [ESP32 devices](http://www.nodemcu.com/index_en.html). Google translate might help a bit to purchase these but if you wish you can also purchase these off the well known [Aliexpress.com](aliexpress.com) but the selection is more limited and not up to date.
+
+![Example Taobao device](https://raw.githubusercontent.com/BenEdridge/blog-usa/master/images/2018/07/taobao_esp8266.png)
+
+Yes you read that right. **16 Chinese Yuan!**  That is about 2 USD. A more powerful ESP32 board with OLED and onboard USB TTL ends up being about 8 USD. Espressif has huge support and there is no lack of software, blog posts and research. The perfect experimentation platform for IoT, so lets dig right in!
+
+# Generic IoT setup
+
+Firstly, I needed to have some way of connecting to the device. This required the usual download of drivers, unloading/loading of kernel modules and debugging of serial access. This is not the focus of post but I have a few resources in the final section if you need some tips.
```

Comment:
> This is not the focus of post but I have a few resources in the final section if you need some tips.

Maybe change it to `Although not the focus of this post,  I have a few resources in the final section if you need some tips.`